enhancements/api-review/add-new-CRD-ImageDigestMirrorSet-and-ImageTagMirrorSet-to-config.openshift.io.md

```diff
@@ -0,0 +1,426 @@
+For users with upgraded clusters that suppport the `ImageTagMirrorSet` CRD, if they had ICSP objects, the ICSP objects still take
+effect. The user will need to configure mirrors using `ImageTagMirrorSet` CRD on the upgraded cluster to use tags.
+The MCO will consume the ImageTagMirrorSet object. Once this is done, the images can be pulled from the mirrors without the digest referenced.
+
+#### As a user， I would like to pull image from mirrors and block the repository in the pull spec
+
+The user can set neverContactSource to true in ImageTagMirrorSet or ImageDigestMirrorSet, depending on which CR the user uses to configure the mirror. The image will still use mirrors to pull imgae, but the pull will not be redirceted to the pull spec if the
+mirrors fail.
+
+#### As a user using ICSP, I would like to use ICSP pull images using digest by default from mirrors
+
+The user can still define ICSP CR before its deprecation. After the deprecation of ICSP, the user will need to use
+`ImageDigestMirrorSet` for digest pull specification requirement.
+
+### API Extensions
+- Adds new CRD ImageDigestMirrorSet to config.openshift.io/v1.
+- Adds new CRD ImageTagMirrorSet to config.openshift.io/v1
+- Impacts on existing resources(The impacts are in the "Operational Aspects
+of API Extensions" section)
+  - Components currently using ImageContentSourcePolicy should upgrade to use ImageDigestMirrorSet.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Implementing this enhancement requires changes in following components (should use underlying code from c/image/pkg/sysregistriesv2 or similary-centralized code):
+- [containers/images](https://github.com/containers/image): registries.conf needs change its mirror setting to make digests only configurable for mirrors registry, not for the main registry. This can be achieved by having a boolean for each mirror table, or having separated mirror lists: a list of mirrors that allow for tag specification,
+and a list of mirrors require digest specification, or have different boolean value.
+- [openshift/api/config/v1](https://github.com/openshift/api/tree/master/config/v1): definition the schema and API of the ImageDigestMirrorSet and ImageTagMirrorSet.
+- [openshift/client-go](https://github.com/openshift/client-go), [openshift/cluster-config-operator](https://github.com/openshift/cluster-config-operator/pull/220): rebase the openshift/api version in these repositories to apply the new CRD
+to the cluster.
+- [openshift/runtime-utils/pkg/registries](https://github.com/openshift/runtime-utils/tree/master/pkg/registries): helper functions to edit registries.conf.
+- [openshift/machine-config-operator](https://github.com/openshift/machine-config-operator): MCO needs watch for the ImageDigestMirrorSet and ImageTagMirrorSet. The machine-config-operator/pkg/controller/container-runtime-config controller needs to operate the ImageDigestMirrorSet and ImageTagMirrorSet CRDs.
```

Comment:
Can we just clarity the first sentence, the container runtime config controller that currently watching ICSP will also watch the new objects. MCO is a little imprecise for this section, since it's implementation details (same with line 303 below). :)